addon/base-record-array.js

```diff
@@ -166,6 +212,15 @@ if (CUSTOM_MODEL_CLASS) {
 } else {
   BaseRecordArray = class BaseRecordArray extends EmberObject.extend(MutableArray) {
     // public RecordArray API
+    static create(...args) {
+      let instance = super.create(...args);
+
+      if (PROXY_MODEL_CLASS) {
+        return new Proxy(instance, new BaseRecordArrayProxyHandler());
+      }
+
+      return instance;
+    }
```

Comment:
I don't think we want this; this is an error condition.

`PROXY_MODEL_CLASS` requires `CUSTOM_MODEL_CLASS`